jdaviz/configs/imviz/helper.py

```diff
@@ -26,6 +41,157 @@ def load_data(self, data, parser_reference=None, **kwargs):
+
+    @property
+    def autocut_options(self):
+        raise NotImplementedError
+
+    @property
+    def cuts(self):
+        raise NotImplementedError
+
+    @cuts.setter
+    def cuts(self, val):
+        raise NotImplementedError
+
+    @property
+    def colormap_options(self):
+        raise NotImplementedError
+
+    def set_colormap(self, cmap):
+        raise NotImplementedError
+
+    @property
+    def click_center(self):
+        return False  # No way to do this right now
+
+    @click_center.setter
+    def click_center(self, val):
+        raise NotImplementedError
+
+    @property
+    def click_drag(self):
+        return False  # No way to do this right now
+
+    @click_drag.setter
+    def click_drag(self, value):
+        raise NotImplementedError
+
+    @property
+    def scroll_pan(self):
+        return False  # No way to do this right now
+
+    @scroll_pan.setter
+    def scroll_pan(self, value):
+        raise NotImplementedError
+
+    def save(self, filename):
+        raise NotImplementedError
+
+    # astrowidgets ducktyping (not MVP)
+    #
+    # NOTES:
+    # * logger is excluded because it is too Ginga-specific.
+    # * cursor is excluded because coordinate info is in a plugin.
+
+    @property
+    def image_width(self):
+        raise NotImplementedError
+
+    @image_width.setter
+    def image_width(self, value):
+        raise NotImplementedError
+
+    @property
+    def image_height(self):
+        raise NotImplementedError
+
+    @image_height.setter
+    def image_height(self, value):
+        raise NotImplementedError
+
+    @property
+    def pixel_offset(self):
+        return 0  # No way to customize right now
+
+    # Loaders
+
+    def load_fits(self, fitsorfn, numhdu=None, memmap=None):
+        raise NotImplementedError
+
+    def load_nddata(self, nddata):
+        raise NotImplementedError
+
+    def load_array(self, arr):
+        raise NotImplementedError
+
+    # Markers
+
+    RESERVED_MARKER_SET_NAMES = ['all']
+
+    @property
+    def is_marking(self):
+        return False  # No way to mark right now
+
+    def start_marking(self, marker_name=None, marker=None):
+        raise NotImplementedError
+
+    def stop_marking(self, clear_markers=False):
+        raise NotImplementedError
+
+    @property
+    def marker(self):
+        return {}  # No way to mark right now
+
+    @marker.setter
+    def marker(self, val):
+        raise NotImplementedError
+
+    def get_markers(self, x_colname='x', y_colname='y',
+                    skycoord_colname='coord', marker_name=None):
+        raise NotImplementedError
+
+    def add_markers(self, table, x_colname='x', y_colname='y',
+                    skycoord_colname='coord', use_skycoord=False,
+                    marker_name=None):
+        raise NotImplementedError
+
+    def remove_markers(self, marker_name=None):
+        raise NotImplementedError
+
+    def reset_markers(self):
+        raise NotImplementedError
```

Comment:
I think this *is* MVP in the sense that this might be what we meant by "regions", given that the regions of the sort glue uses that word for don't scale to thousands... (I thought there was an issue about this... but am not sure now?)

Reply:
See #662 for an update to this comment.

Reply:
But surely, not all the marker methods are MVP?

For instance, how are we supposed to do Glue scatter plot interactively?

Reply:
Ah, okay. Looking at #662 , you only want `add_markers`, `remove_markers`, and *maybe* `reset_markers`, correct?